src/graph.py

```diff
@@ -86,3 +86,24 @@ def adjacent(self, val1, val2):
         if val2 in self[val1]:
             return True
         return False
+
+    def depth_first_traversal(self, start_val, path=[]):
```

Comment:
This is great!  Its worth knowing the iterative version of this and the trade offs of recursion vs. iteration (e.g. memory efficiency of recursion).